x-pack/plugins/security_solution/public/detections/components/rules/step_about_rule/index.tsx

```diff
@@ -96,38 +95,37 @@ const StepAboutRuleComponent: FC<StepAboutRuleProps> = ({
     options: { stripEmptyFields: false },
     schema,
   });
+  const { submit } = form;
 
   const onSubmit = useCallback(async () => {
     if (setStepData) {
       setStepData(RuleStep.aboutRule, null, false);
-      const { isValid, data } = await form.submit();
+      const { isValid, data } = await submit();
       if (isValid) {
         setStepData(RuleStep.aboutRule, data, isValid);
         setMyStepData({ ...data, isNew: false } as AboutStepRule);
       }
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [form]);
+  }, [setStepData, submit]);
 
   useEffect(() => {
-    const { isNew, ...initDefaultValue } = myStepData;
-    if (defaultValues != null && !deepEqual(initDefaultValue, defaultValues)) {
-      const myDefaultValues = {
-        ...defaultValues,
-        isNew: false,
-      };
-      setMyStepData(myDefaultValues);
-      setFieldValue(form, schema, myDefaultValues);
```

Comment:
@sebelga we have this pattern in several of these "rule creation step" forms. Do you intend to change the others after we verify this one?

Reply:
@rylnd We will have to. `setFieldValue` was probably a hack that you had to create because of an unexpected behaviour on the form lib. Now that all the handlers are memoized and the `form` object too, this hack is not necessary anymore. I will review it tomorrow with Patryk.

Reply:
@sebelga yes, absolutely! Very excited to clean things up once this is merged 🙏

Reply:
++ @rylnd's sentiment! 🙂 Between this and https://github.com/elastic/kibana/issues/60602 we'll really be able to clean things up on our end. Thanks for all the work here @sebelga!

Reply:
Thanks @spong ! And good that you bring https://github.com/elastic/kibana/issues/60602 back to my attention 👍